extensions-contrib/compressed-bigdecimal/src/main/java/org/apache/druid/compressedbigdecimal/CompressedBigDecimal.java

```diff
@@ -266,15 +262,27 @@ protected static <S> int signumInternal(int size, S rhs, ToIntBiFunction<S, Inte
    * @see java.lang.Comparable#compareTo(java.lang.Object)
    */
   @Override
-  public int compareTo(CompressedBigDecimal<T> o)
+  public int compareTo(CompressedBigDecimal o)
   {
 
-    if (this.equals(o)) {
+    if (super.equals(o)) {
```

Comment:
Why the switch to super?  Will `Number`'s `.equals` do good things?

Reply:
oops, it was a hangover from when i was changing this.equals() and it wanted to do identity check (and java won't let you use == for Number identity check due to auto-unboxing ambiguity and CBD extends Number...

will revert

Reply:
actually i need to leave it in. the update does change equals mostly  because this overrode compareTo() without equals so I did a quick simple override of equals/hashcode to delegate to toBigDecimal.

it won't perf well but it keeps it functionally correct